### PR TITLE
PrepareRepairItemEvent and RepairItemEvent - BUKKIT-3019

### PR DESCRIPTION
**Pull request by natemort**
Adds PrepareRepairItemEvent and RepairItemEvent.

Addresses BUKKIT-3019.  Use case can be found at the Leaky Ticket(https://bukkit.atlassian.net/browse/BUKKIT-3019).  

CraftBukkit PR:  https://github.com/Bukkit/CraftBukkit/pull/936

### CODE REVIEW

File: src/main/java/org/bukkit/event/inventory/PrepareRepairItemEvent.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ */
+public class PrepareRepairItemEvent extends InventoryEvent{
+
+    private static final HandlerList handlers = new HandlerList();
+    private final int expCost;
+    private AnvilInventory inventory;
+    private String name;
+    private ItemStack result;
+    private int itemCost;
+
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/inventory/PrepareRepairItemEvent.java:19]: Wouldn't some of these, at least inventory, be final?

**Comment by Wolvereness** [comment]
I don't understand the decision to split the two events, yet have them extend each other contrary to proper naming / inheritance conventions?

**Comment by turt2live** [comment]
It appears as though they wanted to mirror functionality of (Prepare)CraftItemEvent

**Comment by natemort** [comment]
What @turt2live said is correct, and I designed it to mirror the functionality of CraftItemEvent, as that's a very similar case.  If you compare these two events to those two events, they are very similar, with the primary differences being found in PrepareRepairItemEvent.

@Wolvereness  There are a few things I need to change before this gets pulled, so I'll make another PR once I've changed them.

**Pull request closed without merging**